components/service/nimbus/src/main/java/mozilla/components/service/nimbus/ext/Nimbus.kt

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.nimbus.ext
+
+import mozilla.components.service.nimbus.NimbusApi
+import mozilla.components.support.base.log.logger.Logger
+import org.mozilla.experiments.nimbus.Variables
+
+/**
+ * Gets the branch name of an experiment acting on the feature given [featureId], and transforms it
+ * with given closure.
+ *
+ * You are probably looking for [withVariables].
+ *
+ * If we're enrolled in an experiment, the transform is passed the branch id/slug as a `String`.
+ *
+ * If we're not enrolled in the experiment, or the experiment is not valid then the transform
+ * is passed a `null`.
+ */
+fun <T> NimbusApi.withExperiment(featureId: String, transform: (String?) -> T): T {
+    return transform(withExperiment(featureId))
+}
+
+/**
+ * The synonym for [NimbusApi.getExperimentBranch] to complement [withExperiment].
+ *
+ * Short-hand for [NimbusApi.getExperimentBranch].
+ */
+@Suppress("TooGenericExceptionCaught")
+fun NimbusApi.withExperiment(featureId: String) =
+    try {
+        getExperimentBranch(featureId)
+    } catch (e: Throwable) {
+        Logger.error("Failed to getExperimentBranch($featureId)", e)
+        null
+    }
```

Comment:
This is also deprecated, unless it's being used by the opt-out screen or something I would prefer to remove it.

Reply:
Agreed. If you want to use `getExperimentBranch`, then use it directly, and with an `experimentSlug` rather than a `featureId`.

TBC: this is from a time before the `Variables` API, and was a transitionary phase while we still weren't sure what the Nimbus API should look like.